lib/utils/cloud.ts

```diff
@@ -106,3 +106,105 @@ export const getDeviceAndMaybeAppFromUUID = _.memoize(
 	// Memoize the call based on UUID
 	(_sdk, deviceUUID) => deviceUUID,
 );
+
+/**
+ * Download balenaOS image for the specified `deviceType`.
+ * `OSVersion` may be one of:
+ *  - exact version number,
+ *  - valid semver range,
+ *  - `latest` (includes pre-releases),
+ *  - `default` (excludes pre-releases if at  least one stable version is available),
+ *  - `recommended` (excludes pre-releases, will fail if only pre-release versions are available),
+ *  - `menu` (will show the interactive menu )
+ * If not provided, OSVersion will be set to `default`
+ *
+ * @param deviceType
+ * @param outputPath
+ * @param OSVersion
+ */
+export async function downloadOSImage(
+	deviceType: string,
+	outputPath: string,
+	OSVersion?: string,
+) {
+	const unzip = await import('node-unzip-2');
+	const fs = await import('fs');
+	const manager = await import('balena-image-manager');
```

Comment:
I'd also move this directly before it's used, that way if `resolveOSVersion` fails we can avoid it, it also means we'll print the warning asap which should be better UX rather than having a slight delay